src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -69,13 +69,24 @@ public function createShopListQueryBuilder(
         ChannelInterface $channel,
         TaxonInterface $taxon,
         string $locale,
-        array $sorting = []
+        array $sorting = [],
+        bool $reproduceTaxonomyTree
     ): QueryBuilder {
         $queryBuilder = $this->createQueryBuilder('o')
             ->addSelect('translation')
             ->innerJoin('o.translations', 'translation', 'WITH', 'translation.locale = :locale')
-            ->innerJoin('o.productTaxons', 'productTaxon')
-            ->andWhere('productTaxon.taxon = :taxon')
+            ->innerJoin('o.productTaxons', 'productTaxon');
+
+        if ($reproduceTaxonomyTree) {
+            $queryBuilder->orWhere('productTaxon.taxon = :taxon');
```

Comment:
I wonder if it generates the correct query (something like `SELECT * FROM xd WHERE taxon = 1 OR taxon = 2 AND channel = 3 AND enabled = true` would not be correct, we expect it to be `SELECT * FROM xd WHERE (taxon = 1 OR taxon = 2) AND channel = 3 AND enabled = true`).